website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+
+In the image, you can also see where the four vertices go in relation to the quad. While you won't *usually* be defining vertices yourself, it's still handy to know how it works.
+
+### Creating the buffer
+Now that we've got our vertex data, let's create the buffer!
+
+At the top of your class, add the following:
+
+```cs
+private static uint _vbo;
+```
+
+Then, in your `OnLoad` method, under where the vertices are defined, add:
+
+```
+_vbo = _gl.GenBuffer();
+_gl.BindBuffer(BufferTargetARB.ArrayBuffer, _vbo);
+```
+
+Much like the VAO, we need to first generate the buffer, and then bind it. Unlike vertex arrays, however, buffers need to be bound to a **target**. This allows you to bind buffers to different targets at the same time. Some of the more common targets include:
+
+* `ArrayBuffer` - The vertex buffer target (which is what we're using here).
+* `ElementArrayBuffer` - The element array buffer target (which you'll see later).
+* `UniformBuffer` - A uniform buffer, not used in this tutorial, but we'll get to it in a later tutorial.
+
+In this case, we're binding to `ArrayBuffer` since we're creating a vertex buffer.
+
+### Filling our buffer with data
+Let's fill our buffer with some data! Before we do that though, you need to be aware of `unsafe` in C#.
+
+#### Unsafe C#
+Silk.NET heavily uses `unsafe` code. Don't worry, this won't make your computer explode, however it does exit out of the "safe" managed environment of C#, and enters a realm where undefined behaviour, segmentation faults, and strange results are more likely to occur if you are not careful.
+
+<?# Info "If you wish to use `Span` instead, and remain in `safe` mode, Silk.NET does support these too. However, I will be using `unsafe` in this tutorial instead, as this is both what I personally use, as well as what the samples use." /?>
+
+Unsafe mode is not enabled by default, so we need to enable it. To enable it, add `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>` to the `PropertyGroup` in your project's `.csproj` file. You also **MUST** add `unsafe` to your `OnLoad` and `OnRender` methods, like so:
+
+```cs
+public static unsafe void OnLoad() { ... }
+```
+
+If you forget to do either of these, you'll get compile errors. While our `OnRender` method does not contain any unsafe code yet, it will when we draw to the screen later, so it's best just to do it now so you don't forget to do it later.
+
+Now let's fill the buffer!
+
+Add the following after you create the buffer:
+
+```cs
+fixed (float* buf = vertices)
+    _gl.BufferData(BufferTargetARB.ArrayBuffer, (nuint) (vertices.Length * sizeof(float)), buf, BufferUsageARB.StaticDraw);
+```
+
+Let's go over what's going on here. First, we `fix` the vertices data. This prevents the garbage collector from moving it around, so we can take a pointer to the data. We tell it the target we want, the `ArrayBuffer` in this case, give it the data length, the buffer pointer, and we choose `StaticDraw` for our usage.
```

Comment:
```suggestion
Let's go over what's going on here. First, we `fix` the vertex data. This prevents the garbage collector from moving it around, so we can take a pointer to the data. We then call `_gl.BufferData` and tell it the target we want, `ArrayBuffer` in this case, give it the data length (measured in bytes), the buffer pointer, and we choose `StaticDraw` for our usage hint.
```

Reply:
Should we elaborate on what `StaticDraw` means? perhaps a list with small descriptions should be somewhere?

Reply:
Yeah, I'll add a small description of what static draw means, along with dynamic draw and stream draw.

I don't remember the others and I've never used them so I'll probably skip their explanation for this tutorial.

The funny thing is that most GL drivers don't even care what usage hint you give them, it's pretty much all the same now (from what I've heard).